app/controllers/devices_controller.rb

```diff
@@ -0,0 +1,42 @@
+class DevicesController < ApplicationController
+  # Devices can only be created for authenticated users.
+  # This replaces the Authenticated Users Pusher Beams solution.
+  # See: https://github.com/forem/forem/pull/12419/files#r563906038
+  before_action :authenticate_user!, only: [:create]
+
+  rescue_from ActiveRecord::ActiveRecordError do |exc|
+    render json: { error: exc.message, status: 422 }, status: :unprocessable_entity
+  end
+
+  def create
+    device = Device.find_or_create_by(device_params)
+    if device.persisted?
+      head :ok
+    else
+      render json: { error: device.errors_as_sentence }, status: :bad_request
+    end
+  end
+
+  def destroy
+    device = Device.find_by(device_params)
+    render json: { error: "Not Found", status: 404 }, status: :not_found unless device
```

Comment:
Mmm I wonder if this will double render as there's no `return` statement from the method.

Reply:
Oh you are right..I think I'm going senile 😆